sdk/eventhub/event-hubs/src/eventData.ts

```diff
@@ -284,3 +288,43 @@ export interface ReceivedEventData {
     [key: string]: any;
   };
 }
+
+/**
+ * Converts any Date objects into a number representing date.getTime().
+ * Recursively checks for any Date objects in arrays and objects.
+ * @internal
+ */
+function convertDatesToNumbers<T = unknown>(thing: T): T {
+  // fast exit
+  if (!isDefined(thing)) return thing;
+
+  // When 'thing' is a Date, return the number representation
+  if (
+    typeof thing === "object" &&
+    objectHasProperty(thing, "getTime") &&
+    typeof thing.getTime === "function"
+  ) {
+    return thing.getTime();
+  }
+
+  /*
+    Examples:
+    [0, 'foo', new Date(), { nested: new Date()}]
+  */
+  if (Array.isArray(thing)) {
+    return (thing.map(convertDatesToNumbers) as unknown) as T;
+  }
+
+  /*
+    Examples:
+    { foo: new Date(), children: { nested: new Date() }}
+  */
+  if (typeof thing === "object" && isDefined(thing)) {
```

Comment:
I think you established that 'thing' can't be undefined above on line 299.

Reply:
I did...this was purely to keep TypeScript happy. Otherwise I can use the dreaded `!` non-null assertion.